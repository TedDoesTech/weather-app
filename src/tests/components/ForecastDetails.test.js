import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: "1111111",
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: "30",
    wind: {
      speed: 17,
      direction: "stub",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        humidity={validProps.humidity}
        wind={validProps.wind}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
