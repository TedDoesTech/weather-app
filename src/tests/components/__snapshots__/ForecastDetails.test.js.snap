// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Invalid Date
    </div>
    <div
      class="forecast-details__icon"
      data-testid="forecast-icon"
    >
      <img
        alt="stubIcon"
      />
    </div>
    <div
      class="forecast-details__mintemperature"
    >
      12°C
    </div>
    <div
      class="forecast-details__maxtemperature"
    >
      22°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      30
    </div>
     
    <div
      class="forecast-details__wind"
    >
      17&stub
    </div>
    <div
      class="forecast-details__description"
    >
      Stub description
    </div>
  </div>
</DocumentFragment>
`;
